lib/puppet/settings/ini_file.rb

```diff
@@ -34,6 +34,13 @@ def append(line)
     @lines << line
   end
 
+  def delete(section, name)
+    entry = setting(section, name)
+    if entry
+      @lines.reject! { |l| l == entry }
```

Comment:
Each line in `@lines` has a reference to the previous line, so we'll need to make sure that the line after the one we want to remove gets updated so it's no longer pointing to the removed line.

Reply:
For clarity: It's not causing a problem when only deleting a line, but would if deleting and adding within the same Manipulator as adding uses the `previous` to find where it should do certain things.